app/src/main/java/com/example/ohthmhyh/activities/UpdateHabitEventActivity.java

```diff
@@ -285,7 +337,8 @@ public void final_create_habit(View view) {
             bitmap = BitmapFactory.decodeResource(getResources(), R.drawable.lol_pic);
         }
 
-        // TODO: Associate the HabitEvent with an actual Habit.
+        habit=checkHabitNull(habit);
```

Comment:
Just do the check in this method (move the code from `checkHabitNull` to here and remove the `checkHabitNull` method).

Reply:
Done